components/Admin/Rings/TeamTournamentRings.tsx

```diff
@@ -0,0 +1,290 @@
+              </LoadingButton>
+              <p>---------</p>
+            </div>
+          ))}
+          <form onSubmit={createAssignment}>
+            <div>
+              <label>
+                <strong>Metsästäjä:</strong>
+                <select
+                  name="assignments"
+                  onChange={(e) => setNewHunter(e.target.value)}
+                >
+                  <option value={""}>--</option>
+                  {teams.map((team) => (
+                    <option key={team.id} value={team.id}>
+                      {getTeamName(team.id)}
+                    </option>
+                  ))}
+                </select>
+              </label>
+            </div>
+            <div>
+              <label>
+                Kohde
+                <select
+                  name="assignments"
+                  onChange={(e) => setNewTarget(e.target.value)}
+                >
+                  <option value={""}>--</option>
+                  {teams.map((team) => (
+                    <option key={team.id} value={team.id}>
+                      {getTeamName(team.id)}
+                    </option>
+                  ))}
+                </select>
+              </label>
+            </div>
+            <LoadingButton
+              type="submit"
+              sx={{
+                fontFamily: "tahoma",
+                fontWeight: "bold",
+                textTransform: "none",
+                letterSpacing: "normal"
+              }}
+              loading={isCreatingAssignment}
+            >
+              Luo uusi toimeksianto
+            </LoadingButton>
+          </form>
+        </div>
+      )}
+    </div>
+  );
+};
+
+export const TeamTournamentRings = ({
+  tournament,
+  rings,
+  teams,
+  setRings
+}: {
+  tournament: Tournament;
+  rings: any[];
+  teams: Team[];
+  setRings;
+}): JSX.Element => {
+  const [newRing, setNewRing] = useState([]);
+  const [showForm, setShowForm] = useState(false);
+  const [isCreatingRing, setIsCreatingRing] = useState(false);
+
+  if (!teams) return;
+
+  const createRing = async (event) => {
+    event.preventDefault();
+    const readyRing = {
+      assignments: Object.values(newRing),
+      name: event.target.ringName.value,
+      tournamentId: tournament.id
+    };
+
+    if (
+      readyRing.assignments.length == 0 ||
+      readyRing.assignments.map((a) => a.targetTeamId).includes("")
+    ) {
+      return;
+    }
+
+    setIsCreatingRing(true);
+
+    const res = await fetch("/api/team-rings/create", {
+      method: "POST",
+      body: JSON.stringify(readyRing)
+    });
+    const createdRing = await res.json();
+    setShowForm(false);
+    setRings(rings.concat(createdRing));
+    setNewRing([]);
+    setIsCreatingRing(false);
+  };
+
+  const handleRingChange = (id, event) => {
+    const assignment = newRing.find(
+      (assignment) => assignment.huntingTeamId === id
+    );
+
+    const teamId = event.target.value;
+
+    if (!assignment) {
```

Comment:
Tässä vois ns. oletus-/perushaarana olla positiivinen keissi, eli ensin käsitellään `if (assigment)`

Reply:
Noin lähtökohtaisesti on ihan ok konventio käsitellä if-elseissä "positiivinen"/true-keissi ensin, jos molemmissa haaroissa tehdään jotain merkittävää, eikä sillä not-keissillä ole jotenkin erityisen painavaa semantiikkaa.